From 33e88f22353aeb285e108579ede5f64f5724631b Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Wed, 2 Dec 2020 11:19:24 +0100
Subject: [PATCH 013/954] feat(stm32mp1): enable MCE driver for STM32MP13

Enabled as BL2 sources.
Add MCE base address and key size in platform definitions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 41e70647bda5d7642210cbdb7a2c44094a6cd4af feat(stm32mp1): enable MCE driver for STM32MP13]
Change-Id: Icb7d9d7a56806a86e6d0a76640a375a5b5e88a2a
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp1/platform.mk    | 4 ++++
 plat/st/stm32mp1/stm32mp1_def.h | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/plat/st/stm32mp1/platform.mk b/plat/st/stm32mp1/platform.mk
index f697293f74..b37eb70a5c 100644
--- a/plat/st/stm32mp1/platform.mk
+++ b/plat/st/stm32mp1/platform.mk
@@ -244,6 +244,10 @@ endif
 BL2_SOURCES		+=	drivers/st/crypto/stm32_hash.c				\
 				plat/st/stm32mp1/bl2_plat_setup.c
 
+ifeq ($(STM32MP13),1)
+BL2_SOURCES		+=	drivers/st/mce/stm32_mce.c
+endif
+
 ifeq (${TRUSTED_BOARD_BOOT},1)
 ifeq ($(STM32MP13),1)
 BL2_SOURCES		+=	drivers/st/crypto/stm32_pka.c
diff --git a/plat/st/stm32mp1/stm32mp1_def.h b/plat/st/stm32mp1/stm32mp1_def.h
index 6530957c31..0529c787c6 100644
--- a/plat/st/stm32mp1/stm32mp1_def.h
+++ b/plat/st/stm32mp1/stm32mp1_def.h
@@ -565,6 +565,14 @@ static inline uintptr_t tamp_bkpr(uint32_t idx)
  ******************************************************************************/
 #define DDRPHYC_BASE			U(0x5A004000)
 
+/*******************************************************************************
+ * STM32MP1 MCE
+ ******************************************************************************/
+#if STM32MP13
+#define MCE_BASE			U(0x58001000)
+#define MCE_KEY_SIZE_IN_BYTES		U(16)
+#endif
+
 /*******************************************************************************
  * STM32MP1 IWDG
  ******************************************************************************/
-- 
2.53.0

